pkgs/development/python-modules/esper/default.nix

```
@@ -0,0 +1,20 @@
+{
+  buildPythonPackage,
+  fetchFromGitHub,
+  flit-core
+}:
+buildPythonPackage rec {
+  pname = "esper";
+  version = "3.2";
+  pyproject = true;
+
```

```suggestion
  pyproject = true;

  disabled = pythonOlder "3.8";
```

see https://github.com/benmoran56/esper/blob/v3.2/pyproject.toml#L12

---

This is what upstream says. Personally I limit new packages to the oldest Python releases which we currently ship to ensure that I don't have to fix issues for older releases and that the tests pass on our Python releases but this is up to the maintainer to decide.

---

I am not convinced about keeping this bit of information as it affects versions of python that are end-of-life and aren't really a concern on the nixpkgs side (since we only have python311 & python312)

---

> since we only have python311 & python312

(39, 310, 311, 312)

---

> > since we only have python311 & python312
> 
> (39, 310, 311, 312)

I was talking with weija and he said me the *only the support for Python 3.11 and 3.12 is continuously guaranteed*. Also, it seems that python39Packages` is not searchable on search.nixorg.org?
What sure is, that there is not support for `python38Packages`

---

> ...as it affects versions of python that are end-of-life ...

If something is end of life it doesn't mean that everybody stop using or depending on it. We still have limited Python 2.7 support because there is a need for that. Also, people do/have to do all kind of wired stuff to keep their Python things running. 

> ...aren't really a concern on the nixpkgs side (since we only have python311 & python312)

On `master` you can still use `python39Packages` or `python39Packages` (see https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/interpreters/python/default.nix#L45) without any kind of additional work:

```bash
$ nix-build -A python39Packages.esper
this derivation will be built:
  /nix/store/nag0wfkqxw25qrjz7lab63dblgp8vc0w-python3.9-esper-3.2.drv
building '/nix/store/nag0wfkqxw25qrjz7lab63dblgp8vc0w-python3.9-esper-3.2.drv'...
[...]
Executing pythonImportsCheckPhase
/nix/store/d4kq9mn69l1ybqxy7b2zkv3m129cq3k5-python3.9-esper-3.2
```

> I was talking with weija and he said me the only the support for Python 3.11 and 3.12 is continuously guaranteed.

Yupp, Python 3.11 and Python 3.12 were the releases which are currently in the focus.

---

Esper readme says *Esper attempts to target all currently supported Python releases (any Python version that is not EOL). *